### PR TITLE
[RDY] Fix sign column auto with minimum break floating window minimal style

### PR DESCRIPTION
**Pull request by spywhere**
<details>
  <summary>As comment by @akinsho in <a href="https://github.com/neovim/neovim/pull/13807#issuecomment-766422813">#13807 (comment)</a></summary>

> I just wanted to give a bit of a heads up that I _think_ that this functionality breaks the way the signcolumn is handled in floating windows. The gif below is an example of me opening telescope whilst using `auto` and then switching to `auto:2-4` which breaks the rendering because of he size of the column.
> 
>   sign_column_bug
> 
>   I'm not entirely sure what exactly is going but it seems to happen only whilst using `auto:min-max` other variations such as `yes:<value>` or just `auto` work fine. It is also reproducible with a minimal init.lua, it's also not specific to `telescope`.

</details>

An introduced option for `signcolumn=auto:[min]-[max]` broke a 'minimal' style of floating windows. 

As of latest master, the current setup for 'minimal' style is to use any `auto` sign column setup
https://github.com/neovim/neovim/blob/8baf7bce2b51e8d3d47b6c70ce054663d927639a/src/nvim/window.c#L678-L682

Considering the following sign column options and its behaviours
- `auto`: Start with minimum _0_ and up to 9 sign columns
- `auto:[max]`: Start with minimum _0_ and up to `max` sign columns
- `auto:[min]-[max]`: Start with minimum `min` and up to `max` sign columns

Only `auto:[min]-[max]` will _require a non-zero minimum sign column_, hence breaking a floating window 'minimal' style.

--------

Also for consideration, from the behavior describe above. Would it be better to change the option to be `yes:[min]-[max]` instead? Since `yes` always require a non-zero sign column anyway and would not need this particular fix.

cc @teto

### CODE REVIEW

**Comment by spywhere** [comment]
@clason Hi, would this PR fall into a `float`/`column` label? And while you're at it, do you know who can review this?

**Comment by clason** [comment]
Can you rebase on master to pull in the CI fix? (While you're at it, tiny nitpick: conventional commit messages are imperative and "positive": they say what the commit actually does, not the previous behavior they fix -- that information is in the linked issue.)

File: src/nvim/window.c [hunk 1]
```diff
@@ -676,7 +676,7 @@ void win_set_minimal_style(win_T *wp)
   }
 
   // signcolumn: use 'auto'
-  if (wp->w_p_scl[0] != 'a') {
+  if (wp->w_p_scl[0] != 'a' || STRLEN(wp->w_p_scl) == 8) {
```
Comment (bfredl) [inline src/nvim/window.c:679]: lets make it >= to forecome eventual support of ten column signcolumn :)

Comment (spywhere) [reply src/nvim/window.c]: Consider it done 😄

**Comment by spywhere** [comment]
Hi @bfredl / @clason , is there anything needed to get this merged?

**Pull request merged**